File: Tyche.Android/bet/src/main/java/com/felipearpa/tyche/bet/domain/PoolGamblerBetResponse.kt
package com.felipearpa.tyche.bet.domain

import java.time.LocalDateTime

data class PoolGamblerBetResponse(
    val poolId: String,
    val gamblerId: String,
    val matchId: String,
    val homeTeamId: String,
    val homeTeamName: String,
    val homeTeamScore: Int?,
    val homeTeamBet: Int?,
    val awayTeamId: String,
    val awayTeamName: String,
    val awayTeamScore: Int?,
    val awayTeamBet: Int?,
    val score: Int?,
    val matchDateTime: LocalDateTime
)

File: Tyche.iOS/Bet/Sources/Bet/Domain/PoolGamblerBetResponse.swift
import Foundation
import Core

struct PoolGamblerBetResponse: Codable {
    let poolId: String
    let gamblerId: String
    let matchId: String
    let homeTeamId: String
    let homeTeamName: String
    let homeTeamScore: Int?
    let homeTeamBet: Int?
    let awayTeamId: String
    let awayTeamName: String
    let awayTeamScore: Int?
    let awayTeamBet: Int?
    let score: Int?
    let matchDateTime: Date
    let isLocked: Bool
}

extension PoolGamblerBetResponse {
    func toPoolGamblerBet() -> PoolGamblerBet {
        return PoolGamblerBet(
            poolId: self.poolId,
            gamblerId: self.gamblerId,
            matchId: self.matchId,
            homeTeamId: self.homeTeamId,
            homeTeamName: self.homeTeamName,
            awayTeamId: self.awayTeamId,
            awayTeamName: self.awayTeamName,
            matchScore: self.homeTeamScore == nil || self.awayTeamScore == nil ?
            nil :
                TeamScore(homeTeamValue: self.homeTeamScore!, awayTeamValue: self.awayTeamScore!),
            betScore: self.homeTeamBet == nil || self.awayTeamBet == nil ?
            nil :
                TeamScore(homeTeamValue: self.homeTeamBet!, awayTeamValue: self.awayTeamBet!),
            score: self.score,
            matchDateTime: self.matchDateTime,
            isLocked: self.isLocked
        )
    }
}
